### DISCUSSION TITLE
lost connection after CONNECT

### DISCUSSION THREAD

**Discussion by RangersMS** [simple-login/app#1508] [Self-hosting question]
After installing it on-prem Simple Login isn't able to send via Postfix. It can connect to it but after Connect connection seems to be dropped by Simple Login.

> lost connection after CONNECT from unknown[169.254.1.3]

FYI: I've changed the docker ip network from the documentation to a custom net (169.254.0.0/16). This change was also done in the main.cf of Postfix for my networks.
`mynetworks = 127.0.0.0/8 [::ffff:127.0.0.0]/104 [::1]/128 169.254.0.0/16`

I followed the [troubleshooting guide](https://github.com/simple-login/app/blob/cbcae31288ff68c036e052fb8d04255a6e276ba5/docs/troubleshooting.md#L1-L0) A1 and A2. Both test were successful. I received the test mail from swaks and I was able to connect from the Simple Login container to the postfix. 

> Dec 29 17:08:20 mail-relay01 postfix/smtpd[13807]: connect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]
Dec 29 17:08:20 mail-relay01 postfix/smtpd[13807]: NOQUEUE: reject: RCPT from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]: 554 5.7.1 <test@mxtoolboxsmtpdiag.com>: Relay access denied; from=<supertool@mxtoolboxsmtpdiag.com> to=<test@mxtoolboxsmtpdiag.com> proto=ESMTP helo=<keeper-us-east-1d.mxtoolbox.com>
Dec 29 17:08:20 mail-relay01 postfix/smtpd[13807]: using backwards-compatible default setting smtpd_relay_before_recipient_restrictions=no to reject recipient "test@mxtoolboxsmtpdiag.com" from client "keeper-us-east-1d.mxtoolbox.com[18.209.86.113]"
Dec 29 17:08:21 mail-relay01 postfix/smtpd[13807]: disconnect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113] ehlo=1 mail=1 rcpt=0/1 quit=1 commands=3/4
Dec 29 17:08:53 mail-relay01 postfix/smtpd[13807]: connect from unknown[169.254.1.3]
Dec 29 17:08:53 mail-relay01 postfix/smtpd[13807]: lost connection after CONNECT from unknown[169.254.1.3]
Dec 29 17:08:53 mail-relay01 postfix/smtpd[13807]: disconnect from unknown[169.254.1.3] commands=0/0
Dec 29 17:12:13 mail-relay01 postfix/anvil[13810]: statistics: max connection rate 1/60s for (smtp:18.209.86.113) at Dec 29 17:08:20
Dec 29 17:12:13 mail-relay01 postfix/anvil[13810]: statistics: max connection count 1 for (smtp:18.209.86.113) at Dec 29 17:08:20
Dec 29 17:12:13 mail-relay01 postfix/anvil[13810]: statistics: max cache size 1 at Dec 29 17:08:20
Dec 29 17:39:05 mail-relay01 postfix/smtpd[14974]: connect from unknown[169.254.1.3]
Dec 29 17:39:33 mail-relay01 postfix/smtpd[14974]: disconnect from unknown[169.254.1.3] quit=1 unknown=0/1 commands=1/2

The Simple Login containers don't create any log entry at that time. The only logs are from the start. After that no log entry is generated. 

PS: The first connection from mxtoolbox.com was to verify I don't have an open relay. This also verifies that Postfix is connected to the PostgreSQL. At the beginning I had issues connecting to the db because of special characters in the password and then Postfix created the following log entry

> Dec 29 13:51:22 mail-relay01 postfix/smtpd[5827]: connect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]
Dec 29 13:51:22 mail-relay01 postfix/trivial-rewrite[5831]: warning: connect to pgsql server localhost: connection to server at "localhost" (::1), port 5432 failed: Connection refused??Is the server running on that host and accepting TCP/IP connections??connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  password authentication failed for user "'simplelogin'"?
Dec 29 13:51:22 mail-relay01 postfix/trivial-rewrite[5831]: warning: relay_domains: pgsql:/etc/postfix/pgsql-relay-domains.cf: table lookup problem
Dec 29 13:51:22 mail-relay01 postfix/trivial-rewrite[5831]: warning: relay_domains lookup failure
Dec 29 13:51:22 mail-relay01 postfix/trivial-rewrite[5831]: warning: relay_domains: pgsql:/etc/postfix/pgsql-relay-domains.cf: table lookup problem
Dec 29 13:51:22 mail-relay01 postfix/trivial-rewrite[5831]: warning: relay_domains lookup failure
Dec 29 13:51:22 mail-relay01 postfix/smtpd[5827]: NOQUEUE: reject: RCPT from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]: 451 4.3.0 <supertool@mxtoolboxsmtpdiag.com>: Temporary lookup failure; from=<supertool@mxtoolboxsmtpdiag.com> to=<test@mxtoolboxsmtpdiag.com> proto=ESMTP helo=<keeper-us-east-1d.mxtoolbox.com>
Dec 29 13:51:22 mail-relay01 postfix/smtpd[5827]: disconnect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113] ehlo=1 mail=1 rcpt=0/1 quit=1 commands=3/4
Dec 29 13:51:53 mail-relay01 postfix/smtpd[5827]: connect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]
Dec 29 13:51:54 mail-relay01 postfix/smtpd[5827]: NOQUEUE: reject: RCPT from keeper-us-east-1d.mxtoolbox.com[18.209.86.113]: 451 4.3.0 <supertool@mxtoolboxsmtpdiag.com>: Temporary lookup failure; from=<supertool@mxtoolboxsmtpdiag.com> to=<test@mxtoolboxsmtpdiag.com> proto=ESMTP helo=<keeper-us-east-1d.mxtoolbox.com>
Dec 29 13:51:54 mail-relay01 postfix/smtpd[5827]: disconnect from keeper-us-east-1d.mxtoolbox.com[18.209.86.113] ehlo=1 mail=1 rcpt=0/1 quit=1 commands=3/4

So I can conclude that Postfix now has a connection to the PostgreSQL db.

**Comment by nguyenkims** [comment]
> I've changed the docker ip network from the documentation to a custom net (169.254.0.0/16)

In the simplelogin.env file, the POSTFIX_SERVER is set to `10.0.0.1`, have you updated this value? 

If the A2 test is successful, that means SimpleLogin containers can send emails to Postfix.

**Reply by RangersMS** [reply]
Thank you very much for looking into it!

Yes I have changed that config
`POSTFIX_SERVER=host.docker.internal`
Have used the hostname `host.docker.internal` which always refers to the IP of the Docker host as described in this [Docker Docs](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host) and this [Medium articel](https://medium.com/@TimvanBaarsen/how-to-connect-to-the-docker-host-from-inside-a-docker-container-112b4c71bc66). 

I can see a connection from the SimpleLogin App when I want to register my user and SimpleLogin tries to send a confirmation mail. It's just that from there on nothing happens and the connection get's disconnected.

> Dec 29 17:39:05 mail-relay01 postfix/smtpd[14974]: connect from unknown[169.254.1.3]
> Dec 29 17:39:33 mail-relay01 postfix/smtpd[14974]: disconnect from unknown[169.254.1.3] quit=1 unknown=0/1 commands=1/2

**Reply by nguyenkims** [reply]
Can you try with the numbered IP address? I suspect that SL containers aren't able to connect to Postfix which is running on the host.
